### DISCUSSION TITLE
I think I'm going to make a 3D physics extension

### DISCUSSION THREAD

**Discussion by Brackets-Coder** [TurboWarp/extensions#1784] [Q&A] [answered]
I've been considering it for a while, and I think there should be a 3D rigidbody/collision extension. I think I'm going to make it. It would be like Box2D, but for 3D. It wouldn't render anything, it wouldn't be a 3D engine, just a physics calculator with a rigid body world and dynamic collision object management. It would be able to detect if an object is touching an other object and list all objects touching a specified object as well as manage static and dynamic rigid body physics, similar to Box2D.

I guess I'm here to see what you think and know if anyone else is making something like this or if it already exists. If so, I won't make it.

**Comment by CubesterYT** [comment, marked as answer]
Doesn't exist, I can assure you that.

**Reply by Brackets-Coder** [reply]
Thanks. I guess I'll start building a basic framework but it might be a few months or longer before a PR appears
